include/compiler/glue-compiler.h

```diff
@@ -22,6 +22,7 @@
 
 #include <spicy/ast/declarations/unit-hook.h>
 #include <spicy/ast/types/unit.h>
+#include <zeek-spicy/autogen/config.h>
```

Comment:
> I think in general it is not a good idea to include autogenerated headers in "public" headers

Why's that? Not sure I see a difference between standard headers and autogenerated headers in this context. (We also don't follow this rule elsewhere for that reason.)

> Could we instead remove this include here and everywhere else and always emit the currently conditional code 

Honestly, I'm not sure it's worth the trouble to change anything here: this is just for 3.x compatibility, which will go out soon and make this all easier / less complex. I'd rather clean it all up then, which will be quite natural to do.

Reply:
Fair enough, I don't feel strongly about this.

> Why's that? 

Depending on the generated header it can introduce churn which can make it harder to cache compilation results. I think in Spicy we try to avoid this as much as possible.

> Honestly, I'm not sure it's worth the trouble to change anything here: ...

I wiped up a patch before suggesting this,

```.patch
diff --git a/include/compiler/glue-compiler.h b/include/compiler/glue-compiler.h
index 5c3ad54..c1a00c0 100644
--- a/include/compiler/glue-compiler.h
+++ b/include/compiler/glue-compiler.h
@@ -22,7 +22,6 @@
 
 #include <spicy/ast/declarations/unit-hook.h>
 #include <spicy/ast/types/unit.h>
-#include <zeek-spicy/autogen/config.h>
 
 #include "driver.h"
 
@@ -63,7 +62,6 @@ struct FileAnalyzer {
     std::optional<UnitInfo> unit; /**< The type of the unit to parse the originator side. */
 };
 
-#ifdef HAVE_PACKET_ANALYZERS
 /** Representation of a Spicy packet analyzer, parsed from an EVT file. */
 struct PacketAnalyzer {
     // Information parsed directly from the *.evt file.
@@ -74,7 +72,6 @@ struct PacketAnalyzer {
     // Computed information.
     std::optional<UnitInfo> unit; /**< The type of the unit to parse the originator side. */
 };
-#endif
 
 /**
  * Representation of an expression computing the value of a parameter passed
@@ -175,9 +172,7 @@ private:
     // Parsers for parts from EVT files.
     glue::ProtocolAnalyzer parseProtocolAnalyzer(const std::string& chunk);
     glue::FileAnalyzer parseFileAnalyzer(const std::string& chunk);
-#ifdef HAVE_PACKET_ANALYZERS
     glue::PacketAnalyzer parsePacketAnalyzer(const std::string& chunk);
-#endif
     glue::Event parseEvent(const std::string& chunk);
 
     /** Computes the missing pieces for all `Event` instances.  */
@@ -203,9 +198,7 @@ private:
     std::vector<glue::Event> _events;                        /**< events parsed from EVT files */
     std::vector<glue::ProtocolAnalyzer> _protocol_analyzers; /**< protocol analyzers parsed from EVT files */
     std::vector<glue::FileAnalyzer> _file_analyzers;         /**< file analyzers parsed from EVT files */
-#ifdef HAVE_PACKET_ANALYZERS
-    std::vector<glue::PacketAnalyzer> _packet_analyzers; /**< file analyzers parsed from EVT files */
-#endif
-    std::vector<hilti::Location> _locations; /**< location stack during parsing EVT files */
+    std::vector<glue::PacketAnalyzer> _packet_analyzers;     /**< file analyzers parsed from EVT files */
+    std::vector<hilti::Location> _locations;                 /**< location stack during parsing EVT files */
 };
 } // namespace spicy::zeek
diff --git a/include/cookie.h b/include/cookie.h
index 66fdab9..e9a28ed 100644
--- a/include/cookie.h
+++ b/include/cookie.h
@@ -14,7 +14,6 @@
 
 #include <hilti/rt/fmt.h>
 
-#include <zeek-spicy/autogen/config.h>
 #include <zeek-spicy/zeek-compat.h>
 
 namespace spicy::zeek::rt {
diff --git a/include/plugin.h b/include/plugin.h
index db2c021..6e18eac 100644
--- a/include/plugin.h
+++ b/include/plugin.h
@@ -12,7 +12,6 @@
 #include <hilti/rt/library.h>
 #include <hilti/rt/types/port.h>
 
-#include <zeek-spicy/autogen/config.h>
 #include <zeek-spicy/zeek-compat.h>
 
 #ifdef ZEEK_SPICY_PLUGIN_USE_JIT
diff --git a/include/zeek-reporter.h b/include/zeek-reporter.h
index eb71edd..9c3f5fb 100644
--- a/include/zeek-reporter.h
+++ b/include/zeek-reporter.h
@@ -11,7 +11,6 @@
 
 #include <string>
 
-#include <zeek-spicy/autogen/config.h>
 #include <zeek-spicy/debug.h>
 
 #include "plugin.h"
diff --git a/src/compiler/glue-compiler.cc b/src/compiler/glue-compiler.cc
index f1e38a4..9e53d66 100644
--- a/src/compiler/glue-compiler.cc
+++ b/src/compiler/glue-compiler.cc
@@ -634,8 +634,8 @@ glue::FileAnalyzer GlueCompiler::parseFileAnalyzer(const std::string& chunk) {
     return a;
 }
 
-#ifdef HAVE_PACKET_ANALYZERS
 glue::PacketAnalyzer GlueCompiler::parsePacketAnalyzer(const std::string& chunk) {
+#ifdef HAVE_PACKET_ANALYZERS
     glue::PacketAnalyzer a;
     a.location = _locations.back();
 
@@ -667,8 +667,10 @@ glue::PacketAnalyzer GlueCompiler::parsePacketAnalyzer(const std::string& chunk)
     }
 
     return a;
-}
+#else
+    throw Unsupported("packet analyzer functionality requires Zeek >= 4.0");
 #endif
+}
 
 glue::Event GlueCompiler::parseEvent(const std::string& chunk) {
     glue::Event ev;
```

Reply:
Well, given that you have already done it, if you want to file that as a separate PR, I won't object.